admin/services/check-complete.service.js

```diff
@@ -25,8 +25,10 @@ checkCompleteService.completeCheck = async function (completedCheck) {
     await pupilDataService.sqlUpdate({ id: pupil.id, pinExpiresAt: moment.utc(), pin: null })
   }
   // Timestamp the request
-  // TODO: This timestamp should be recorded in the application service tier instead
-  completedCheck.receivedByServerAt = moment.utc()
+  const receivedByServerAt = moment.utc()
+
+  // update receivedByServerAt request timestamp
+  await checkDataService.sqlUpdateCheckReceivedByServerAt(completedCheck.data.pupil.checkCode, receivedByServerAt)
```

Comment:
@danwilliams did these changes have any impact on the tests?
I would suspect there are going to be async timeouts unless you spy on this check data service method.

Reply:
Locally, there were no effects... however I do see some timeouts on Circle. I will look into it.